### PR TITLE
riff 0.2.0 (new formula)

### PR DESCRIPTION
**Pull request by fbiville**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/riff.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    (buildpath/"bin").mkpath
+
+    cd "src/github.com/projectriff/riff" do
+      system "make"
+      bin.install "riff"
+    end
+  end
+
+  test do
+    expected_output = <<~EOS
```
Comment (fxcoudert) [inline Formula/riff.rb]: We need a test that exercises the some of the functionality of the app.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/riff.rb (hunk 1 above)
Comment (fbiville) [reply Formula/riff.rb]: That would require an external Kubernetes cluster.

**Comment by fbiville** [comment]
I see that `:optional` and `:recommended` are to be avoided.
`riff` depends on `kubernetes-cli` at runtime. How can I define that in the recipe?

**Comment by SMillerDev** [comment]
>I see that `:optional` and `:recommended` are to be avoided. ...

@fbiville just mark it as "depends_on" without anything special.

**Comment by fbiville** [comment]
@SMillerDev `kubectl` can be installed with `gcloud`, `Docker`... `riff` needs to rely on the `kubectl` in use on the machine, regardless of the way it was set up.

If I specify only `depends_on` without any symbols, `kubernetes-cli` will be downloaded and installed. Isn't `:optional` a better fit?

File: Formula/riff.rb (hunk 1 above)
Comment (fbiville) [reply Formula/riff.rb]: @fxcoudert is the test OK? I'm not sure I can do much more.

**Comment by SMillerDev** [comment]
>@SMillerDev `kubectl` can be installed with `gcloud`, `Docker`... ...

@fbiville other than the audit. Is it optional for usage? Or is there a hard dependency?

**Comment by fbiville** [comment]
@SMillerDev it's a hard dependency indeed. I think most of `kubernetes-cli` users get it from other ways than Homebrew. Happy to make it a regular runtime dependency if you think that's the way to go.

**Comment by SMillerDev** [comment]
> @SMillerDev it's a hard dependency indeed. I think most of `kubernetes-cli` users get it from other ways than Homebrew. Happy to make it a regular runtime dependency if you think that's the way to go.

Since it won't work without it it doesn't really matter if/how people installed alternatives, this is a dependency.

**Comment by fbiville** [comment]
@SMillerDev done, thanks!
Is the test in the recipe OK? I'm afraid that's the best I can do without interacting with a Kubernetes server.

File: Formula/riff.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/riff.rb]: You could check for a failure instead, maybe something that always returns "unautorized?"

Comment (fbiville) [reply Formula/riff.rb]: Done. I'm calling a command with an invalid kube config file.

File: Formula/riff.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class Riff < Formula
+  desc "Function As A Service on top of Knative, riff is for functions"
+  homepage "https://www.projectriff.io/"
+  url "https://github.com/projectriff/riff.git",
+      :tag      => "v0.2.0",
+      :revision => "1ae190ff3c7edf4b375ee935f746ebfd1d8eaf5c"
+
+  depends_on "go" => :build
+  depends_on "make" => :build
```
Comment (fxcoudert) [inline Formula/riff.rb]: Can't it install with macOS system make?

Comment (fbiville) [reply Formula/riff.rb]: Oops, I didn't realize it was shipped by default. Let me remove the dependency.

Comment (fbiville) [reply Formula/riff.rb]: Done

**Comment by fbiville** [comment]
Please let me know if there is anything else I need to do.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Works for me. @SMillerDev any comment?

**Comment by SMillerDev** [comment]
Looks good

**Comment by fxcoudert** [comment]
Thanks @fbiville for your contribution to Homebrew!

**Pull request closed without merging**